### PR TITLE
Add test case for issue #266 ("Invalid read of size 4" in cpu_tb_exec).

### PR DESCRIPTION
**Pull request by practicalswift**

### CODE REVIEW

**Comment by aquynh** [comment]
can you add this to `tests/regress/Makefile` to compile it?
at the same time, please add its binary name (`invalid_read_in_cpu_tb_exec`) to .gitignore.

thanks.

File: tests/regress/invalid_read_in_cpu_tb_exec.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#include <unicorn/unicorn.h>
+
+static void hook_block(uc_engine *uc, uint64_t address, uint32_t size, void *user_data) {
+  printf("hook_block(…)\n");
+}
+
+#define BINARY "\x80\x05\xff\xff\xff\xff\x30\xeb\xf7\x30"
```
Comment (aquynh) [inline tests/regress/invalid_read_in_cpu_tb_exec.c]: please add the assembly of this code (as comment)

**Comment by practicalswift** [comment]
Done. Let me know if any further adjustments are needed! :-)

File: tests/regress/invalid_read_in_cpu_tb_exec.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    printf("uc_open(…) failed\n");
+    return 1;
+  }
+  uc_mem_map(uc, STARTING_ADDRESS, MEMORY_SIZE, UC_PROT_ALL);
+  if (uc_mem_write(uc, STARTING_ADDRESS, BINARY, sizeof(BINARY) - 1)) {
+    printf("uc_mem_write(…) failed\n");
+    return 1;
+  }
+  uc_hook hook;
+  uc_hook_add(uc, &hook, UC_HOOK_BLOCK, hook_block, NULL, 1, 0);
```
Comment (aquynh) [inline tests/regress/invalid_read_in_cpu_tb_exec.c]: the last two params should be cast to uint64_t, to be: (uint64_t)1, (uint64_t)0.
can run this testcase again with this change to see if the problem is still there?

**Comment by practicalswift** [comment]
Done. The issue is still there.

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**